test/framework/options.py

```diff
@@ -1072,6 +1072,49 @@ def check_copied_files():
         self.assertTrue(os.path.exists(copied_toy_cwd))
         self.assertEqual(read_file(copied_toy_cwd), toy_ec_txt)
 
+        # Test --copy-ec coupled with --from-pr
+        all_ecs_pr8007 = [
+            'Arrow-0.7.1-intel-2017b-Python-3.6.3.eb',
+            'bat-0.3.3-fix-pyspark.patch',
+            'bat-0.3.3-intel-2017b-Python-3.6.3.eb',
+        ]
+
+        # test use of `--copy-ec` with `--from-pr` to the cwd
+        test_working_dir = os.path.join(self.test_prefix, 'test_working_dir')
+        mkdir(test_working_dir)
+        change_dir(test_working_dir)
+        args = ['--copy-ec', '--from-pr', '8007']
+        stdout = mocked_main(args)
+        self.assertEqual(stdout, '2 file(s) copied to %s' % test_working_dir)
+        # check that the two easyconfigs exist
+        self.assertTrue(os.path.exists(os.path.join(test_working_dir, all_ecs_pr8007[0])))
+        remove_file(os.path.join(test_working_dir, all_ecs_pr8007[0]))
+        self.assertTrue(os.path.exists(os.path.join(test_working_dir, all_ecs_pr8007[2])))
+        remove_file(os.path.join(test_working_dir, all_ecs_pr8007[2]))
+        # ...but the patch doesn't
+        self.assertFalse(os.path.exists(os.path.join(test_working_dir, all_ecs_pr8007[1])))
+        remove_file(os.path.join(test_working_dir, all_ecs_pr8007[1]))
```

Comment:
Hmm, would it be difficult to fix this? I would expect that the patch gets copied too...

We should also test copying a single easyconfig from a PR where multiple are touched?
Like `eb --copy-ec --from-pr 8007 Arrow-0.7.1-intel-2017b-Python-3.6.3.eb <target_dir>`

Reply:
I can easily copy *all* patches that are alongside an easyconfig, but I don't know how to (easily) filter the ones that are part of a PR

Reply:
Regarding the additional test, it is not going to pick that up, according to the existing logic, it is going to assume that the additional argument is the target path. I'm not sure what logic I could use to get around that, I'd need to know if the filename(s) appear(s) in the PR.

Reply:
Maybe I can do it before I copy the files, if the basename of the target path matches one of the paths then I assume that is what was meant and only copy that single file to the basedir of the target path